src/util/mapbox.js

```diff
@@ -100,18 +101,49 @@ export const normalizeTileURL = function(tileURL: string, sourceURL?: ?string, t
     const suffix = browser.devicePixelRatio >= 2 || tileSize === 512 ? '@2x' : '';
     const extension = browser.supportsWebp ? '.webp' : '$1';
     urlObject.path = urlObject.path.replace(imageExtensionRe, `${suffix}${extension}`);
+    urlObject.path = `/v4${urlObject.path}`;
 
-    replaceTempAccessToken(urlObject.params);
-    return formatUrl(urlObject);
+    return makeAPIURL(urlObject);
 };
 
-function replaceTempAccessToken(params: Array<string>) {
-    for (let i = 0; i < params.length; i++) {
-        if (params[i].indexOf('access_token=tk.') === 0) {
-            params[i] = `access_token=${config.ACCESS_TOKEN || ''}`;
+export const canonicalizeTileURL = function(url: string) {
+    const version = "/v4/";
+
+    const urlObject = parseUrl(url);
+    // Make sure that we are dealing with a valid Mapbox tile URL.
+    // Has to be /v4/, with a valid filename + extension
+    if (!urlObject.path.match(version) || !(urlObject.path.match(imageExtensionRe) || urlObject.path.match('.pbf'))) {
```

Comment:
The logic below depends on `version` matching at the beginning of the path, right? We should probably explicitly require that? Also we should check that the extensions are at the end of the path (I think the image extension regex already does that)?